File: sinal_retangular.py
# coding=utf-8
from numpy import * 
from scipy import * 
from matplotlib.pyplot import *

amostragem = 1000
tmax = 10.0
nyquist = amostragem/2
w1 = 2*pi*nyquist/50
dt = 1.0/amostragem
t = arange(0,tmax,dt)
s = zeros( t.size )
s[0:nyquist] = 1.0
sn =  s[::-1]
tn = -t[::-1]
f = linspace(-amostragem,amostragem,2*t.size)
w = 2*pi*f
S = real( fft(s) )
Sn =  S[::-1]
snp = hstack((sn,s))
tnp = hstack((tn,t))
Snp = hstack((Sn,S))
Snp = Snp/max( abs( Snp ) )

figure(1)
subplot(211),plot(tnp,snp,'k',linewidth=2)
xlabel('$t$ (s)')
ylabel(r'$\Pi(t)$')
xlim(-1,1)
ylim(-0.1,1.1)
grid(True)
subplot(212),plot(w,Snp,'k',linewidth=2)
xlabel(r'$\omega$ (rad/s)')
ylabel(r'FT[$\Pi(t)$]')
xlim(-w1,w1)
ylim(-0.25,1.1)
grid(True)
show()

File: sinal_retangular.m
% Fundamentos de Acústica Submarina 

clear all, close all 

amostragem = 1000;
tmax = 10.0;
nyquist = 500;
w1 = 2*pi*nyquist/50;
dt = 1.0/amostragem;
t = [0:dt:tmax-dt];
s = zeros( 1, length(t) );
s(1:nyquist) = 1.0;
sn =  fliplr( s );
tn = -fliplr( t );
f = linspace(-amostragem,amostragem,2*length(t));
w = 2*pi*f;
S = real( fft(s) );
Sn =  fliplr( S );
snp = [sn,s];
tnp = [tn,t];
Snp = [Sn,S];
Snp = Snp/max( abs( Snp ) );

figure(1)
subplot(211)
plot(tnp,snp,'k','LineWidth',2)
xlabel('t (s)')
ylabel('\Pi(t)')
xlim([-1,1])
ylim([-0.1,1.1])
grid on, box on
subplot(212)
plot(w,Snp,'k','LineWidth',2)
xlabel('\omega (rad/s)')
ylabel('FT[\Pi(t)]')
xlim([-w1,w1])
ylim([-0.25,1.1])
grid on, box on

